### PR TITLE
fix(web): use natural asset order for navigation

### LINKED ISSUE #3091: asset view navigation does not match the timeline
### The bug

Say I have five assets and I am viewing asset #3. Navigating right will navigate to asset #2, and then asset #4. Pretty strange.

In some scenarios it's completely broken. See:

[Screencast from 2023-07-03 00-55-40.webm](https://github.com/immich-app/immich/assets/9749173/2d342d98-492f-4229-bbb8-90065752abeb)

### The OS that Immich Server is running on

N/A

### Version of Immich Server

v1.65.0

### Version of Immich Mobile App

N/A

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
N/A
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by uhthomas**
Fixes: #3091

### CODE REVIEW

File: web/src/lib/stores/asset-interaction.store.ts [hunk 1]
```diff
@@ -64,40 +61,23 @@ function createAssetInteractionStore() {
...
       for (const bucket of _assetGridState.buckets) {
-        if (bucket.assets.length === 0) {
-          nextBucket = bucket.bucketDate;
-          break;
-        }
-      }
-
-      if (nextBucket !== '') {
-        await assetStore.getAssetsByBucket(nextBucket, BucketPosition.Below);
+        await assetStore.getAssetsByBucket(bucket.bucketDate, BucketPosition.Below);
```
Comment (uhthomas) [inline web/src/lib/stores/asset-interaction.store.ts:71]: oops missing condition

**Review by jrasm91 — APPROVED** [review]
LGTM

**Comment by brighteyed** [comment]
I haven't tested it myself yet, but looking into code it seems the issue may still exist.

`_assetGridState.assets` is a flattened list of all the assets for already loaded buckets. In general, buckets are loaded in "random" order if the user clicks on the scrollbar. In this case, we will find the next/prev index, but the asset for this index may not be from the adjacent bucket.

**Comment by brighteyed** [comment]
Yes, I can reproduce the bug on `main` (`2099b04057a161ec44b557abac17763b4e696657`).

Steps to reproduce:
1. Open the main timeline (the first N bucket(s) are loaded now)
2. Click on the scrollbar somewhere near the bottom (some bucket(s) near that location are loaded. Note: `_assetGridState.assets` now contains assets from nonadjacent buckets)
3. Click on any asset to open `AssetViewer`
4. Click the Previous button multiple times. When we reach the end of the bucket(s) loaded in (2), we begin to show assets from buckets loaded in (1) thus missing a lot of assets in intermediate buckets.

**Pull request merged**